### PR TITLE
Update cohort review to match mocks

### PR DESCRIPTION
**Pull request by tjennison-google**
* Move annotations to the list page.
* Add latest/outdated status.
* Add support for condensed cohort summaries.
* Add shared component for tabs.
* Add helper for simple treegrids and move dividers to the top (except the first row) to avoid a dangling divider at the bottom.
* Change style in many places.

Some smaller functionality, style, and text copy changes haven't been done yet and I'll come back to them after tackling some other major features.

### CODE REVIEW

**Comment by tjennison-google** [comment]
image
image
image

File: ui/src/cohortSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+              {g.criteria.length > 1 ? (
+                <span>
+                  <Typography variant="body2em" component="span">
+                    {" where "}
+                  </Typography>
+                  <Typography variant="body2" component="span">
+                    {g.criteria
+                      .slice(1)
+                      .map((c) =>
+                        getCriteriaTitle(c, getCriteriaPlugin(c, g.entity))
```
Comment (marikomedlock) [inline ui/src/cohortSummary.tsx:46]: Does each criteria plugin define its own title, or can they override the default title somehow?

File: ui/src/cohortReview/cohortReviewList.tsx [hunk 2]
```diff
@@ -53,8 +76,8 @@ class ReviewListItem {
     return this.pending?.size ?? this.review?.size;
   }
 
-  created() {
-    return this.pending?.created ?? this.review?.created;
+  lastModified() {
+    return this.pending?.lastModified ?? this.review?.cohort?.lastModified;
```
Comment (marikomedlock) [inline ui/src/cohortReview/cohortReviewList.tsx:80]: This `lastModified` shows the review last modified, with a fallback to the review's cohort revision last modified. Is that correct?

**Review by marikomedlock — APPROVED** [review]

File: ui/src/cohortSummary.tsx (hunk 1 above)
Comment (tjennison-google) [reply ui/src/cohortSummary.tsx:46]: The criteria provides part of the title and it's combined with the criteria type.

File: ui/src/components/treegrid.tsx [hunk 3]
```diff
@@ -254,14 +261,35 @@ export function TreeGrid(props: TreeGridProps) {
             "root",
             "root",
             0,
-            false
+            false,
+            true
```
[the remark this answers is not in the document]
Comment (tjennison-google) [reply ui/src/components/treegrid.tsx:265]: Done.

File: ui/src/cohortReview/cohortReviewList.tsx (hunk 2 above)
Comment (tjennison-google) [reply ui/src/cohortReview/cohortReviewList.tsx:80]: No, this object is to hold either pending reviews that are being created or actually existing reviews. Then they can be displayed in the list without knowing which is which.

**Pull request merged**